### PR TITLE
Adapting to latest Knative

### PR DESCRIPTION
**Pull request by kameshsampath**
Updating for latest Knative and quarkus.

**Check list**:

Your pull request:

- [*] targets the `development` branch
- [*] uses the `999-SNAPSHOT` version of Quarkus
- [x] has tests (`mvn clean test`)
- [*] works in native (`mvn clean package -Pnative`)
- [x] has native tests (`mvn clean verify -Pnative`)
- [*] makes sure the documentation must not be updated
- [* ] links the documentation update pull request (if needed)
https://github.com/quarkusio/quarkus/pull/7100
- [* ] updates or creates the `README.md` file (with build and run instructions)

### CODE REVIEW

File: getting-started-knative/.gitignore [hunk 1]
```diff
@@ -32,4 +32,6 @@ target/
 pom.xml.tag
 pom.xml.releaseBackup
 pom.xml.versionsBackup
-release.properties
\ No newline at end of file
+release.properties
+
+Dockerfile
```
Comment (gsmet) [inline getting-started-knative/.gitignore]: Why is the `Dockerfile` in `.gitignore` but still committed?

File: getting-started-knative/pom.xml [hunk 2]
```diff
@@ -29,7 +27,26 @@
...
+      <scope>test</scope>
+    </dependency>
+    <dependency>
+      <groupId>io.rest-assured</groupId>
+      <artifactId>rest-assured</artifactId>
+      <scope>test</scope>
+    </dependency>
+    <dependency>
+      <groupId>io.quarkus</groupId>
+      <artifactId>quarkus-smallrye-health</artifactId>
```
Comment (gsmet) [inline getting-started-knative/pom.xml]: I would put the test dependencies at the end.

Comment (kameshsampath) [reply getting-started-knative/pom.xml]: done

File: getting-started-knative/.gitignore (hunk 1 above)
Comment (kameshsampath) [reply getting-started-knative/.gitignore]: done

**Comment by kameshsampath** [comment]
@gsmet  - fixed the comments. please review and merge

**Comment by gsmet** [comment]
@kameshsampath can you squash?
@cescoffier could you make a quick scan? I'm not that familiar with all this.

**Comment by kameshsampath** [comment]
@gsmet   and @cescoffier  - I have squashed my commits into one .

**Comment by kameshsampath** [comment]
@cescoffier @gsmet   - will this be merged in  next release, the doc has been merged and this has not so there might be discrepancy

**Pull request merged**